presto-main/src/test/java/com/facebook/presto/execution/TestNodeScheduler.java

```diff
@@ -424,6 +424,24 @@ public void testSplitCount()
         assertEquals(nodeTaskMap.getPartitionedSplitsOnNode(chosenNode), 0);
     }
 
+    @Test
+    public void testMaxTasksPerStage()
+    {
+        TestingTransactionHandle transactionHandle = TestingTransactionHandle.create();
+
+        ImmutableList.Builder<Split> initialSplits = ImmutableList.builder();
+        for (int i = 0; i < 10; i++) {
+            initialSplits.add(new Split(CONNECTOR_ID, transactionHandle, new TestSplitRemote()));
+        }
+
+        Set<Split> splits = new HashSet<>();
+        for (int i = 0; i < 5; i++) {
+            splits.add(new Split(CONNECTOR_ID, transactionHandle, new TestSplitRemote()));
+        }
+        Multimap<InternalNode, Split> assignments = nodeSelector.computeAssignments(splits, ImmutableList.copyOf(taskMap.values()), 2).getAssignments();
```

Comment:
I think this test has to be changed to call the `computeAssignments` multiple times (same as it is done for real query scheduling)

Reply:
Also please make sure the solution works in real deployment. As the scheduling code might get quite tricky.

Reply:
will do.